src/flags.ml

```diff
@@ -1121,6 +1121,503 @@ module Certif = struct
+    | "no" -> `DO_NOT_MINIMIZE
+    | "valid_lustre" -> `VALID_LUSTRE
+    | "concise" -> `CONCISE
+    | _ -> raise (Arg.Bad "Bad value for --minimize_program")
+
+  let minimize_program_default = `DO_NOT_MINIMIZE
+  let minimize_program = ref minimize_program_default
+  let _ = add_spec
+    "--minimize_program"
+    (Arg.String (fun str -> minimize_program := mm_of_string str))
+    (fun fmt ->
+      Format.fprintf fmt
+        "\
+          Minimize the source Lustre program according to the inductive validity core(s) computed@ \
+          \"no\" to disable this feature (default)@ \
+          \"valid_lustre\" to replace useless expressions by a valid node call@ \
+          \"concise\" to replace useless expressions by a '_'\
+        "
+    )
+  let minimize_program () = !minimize_program
+
+  let minimized_program_dir_default = ""
+  let minimized_program_dir = ref minimized_program_dir_default
+  let _ = add_spec
+      "--ivc_output_dir"
+      (Arg.Set_string minimized_program_dir)
+      (fun fmt ->
+         Format.fprintf fmt
+           "Output directory for the minimized programs@ \
+            Default: <INPUT_FILENAME>"
+      )
+  let minimized_program_dir () = !minimized_program_dir
+
+
+  let ivc_precomputed_mcs_default = 0
+  let ivc_precomputed_mcs = ref ivc_precomputed_mcs_default
+  let _ = add_spec
+    "--ivc_precomputed_mcs"
+    (Arg.Set_int ivc_precomputed_mcs)
+    (fun fmt ->
+      Format.fprintf fmt
+        "\
+          When computing all IVCs, determine the cardinality up to which@ \
+          MCS will be computed before starting to compute the IVCs.@ \
+          Correspond to the parameter 'k' of the implementation UMIVC.@ \
+          In particular, the value 0 implements the MARCO algorithm,@ \
+          and the value -1 (infinity) implements the CAMUS algorithm.@ \
+          Default: %n\
+        "
+        ivc_precomputed_mcs_default
+    )
+  let ivc_precomputed_mcs () = !ivc_precomputed_mcs
+
+
+  let ivc_uc_timeout_default = 0
+  let ivc_uc_timeout = ref ivc_uc_timeout_default
+  let _ = add_spec
+    "--ivc_uc_timeout"
+    (Arg.Set_int ivc_uc_timeout)
+    (fun fmt ->
+      Format.fprintf fmt
+        "\
+          Set a timeout for each unsat core check sent to the solver.@ \
+          This setting is ignored if a solver different than Z3 is used.@ \
+          Set to 0 to disable timeout.@ \
+          Default: %n\
+        "
+        ivc_uc_timeout_default
+    )
+  let ivc_uc_timeout () = !ivc_uc_timeout
+
+end
+
+(* Minimal Cut Sets flags. *)
+module MCS = struct
+
+  include Make_Spec (struct end)
+
+  (* Identifier of the module. *)
+  let id = "mcs"
+  (* Short description of the module. *)
+  let desc = "Minimal Cut Sets generation flags"
+  (* Explanation of the module. *)
+  let fmt_explain fmt =
+    Format.fprintf fmt "@[<v>\
+      Kind 2 generates a minimal cut set,@ \
+      that is a minimal subset of the model elements that makes the system unsafe if removed.\
```

Comment:
that is a minimal subset of the model elements whose no satisfaction leads to the violation of a property